tests/components/compit/test_config_flow.py

```diff
@@ -0,0 +1,215 @@
+    """Test user step with invalid authentication."""
+    flow = CompitConfigFlow()
+    flow.hass = hass
+
+    with patch(
+        "homeassistant.components.compit.config_flow.CompitAPI.authenticate",
+        side_effect=InvalidAuth,
+    ):
+        result = await flow.async_step_user(
+            {CONF_EMAIL: "test@example.com", CONF_PASSWORD: "password"}
+        )
+
+    assert result["type"] == FlowResultType.FORM
+    assert result["errors"] == {"base": "invalid_auth"}
+
+
+async def test_async_step_user_cannot_connect(hass: HomeAssistant) -> None:
+    """Test user step with connection error."""
+    flow = CompitConfigFlow()
+    flow.hass = hass
+
+    with patch(
+        "homeassistant.components.compit.config_flow.CompitAPI.authenticate",
+        side_effect=CannotConnect,
+    ):
+        result = await flow.async_step_user(
+            {CONF_EMAIL: "test@example.com", CONF_PASSWORD: "password"}
+        )
+
+    assert result["type"] == FlowResultType.FORM
+    assert result["errors"] == {"base": "cannot_connect"}
+
+
+async def test_async_step_user_unknown_error(hass: HomeAssistant) -> None:
+    """Test user step with unknown error."""
+    flow = CompitConfigFlow()
+    flow.hass = hass
+
+    with patch(
+        "homeassistant.components.compit.config_flow.CompitAPI.authenticate",
+        side_effect=Exception,
+    ):
+        result = await flow.async_step_user(
+            {CONF_EMAIL: "test@example.com", CONF_PASSWORD: "password"}
+        )
+
+    assert result["type"] == FlowResultType.FORM
+    assert result["errors"] == {"base": "unknown"}
+
+
+async def test_async_step_reauth_confirm_success(hass: HomeAssistant) -> None:
+    """Test reauth confirm step with successful authentication."""
+
+    entry = config_entries.ConfigEntry(
+        version=1,
+        domain=DOMAIN,
+        title="Compit",
+        data={CONF_EMAIL: "test@example.com"},
+        source=config_entries.SOURCE_REAUTH,
+        entry_id="1",
+        unique_id="compit_test@example.com",
+        options={},
+        discovery_keys=None,
+        minor_version=0,
+    )
+    hass.config_entries._entries[entry.entry_id] = entry
+
+    flow = CompitConfigFlow()
+    flow.hass = hass
+    flow._get_reauth_entry = Mock(return_value=entry)
+
+    with patch(
+        "homeassistant.components.compit.config_flow.CompitAPI.authenticate",
+        return_value=AsyncMock(
+            return_value=SystemInfo(
+                gates=[Gate(label="Test", code="1", devices=[], id=1)]
+            )
+        ),
+    ):
+        result = await flow.async_step_reauth_confirm(
+            {CONF_PASSWORD: "new_password", CONF_EMAIL: "test@example.com"}
+        )
+
+    assert result["type"] == FlowResultType.ABORT
+    assert result["reason"] == "reauth_successful"
+
+
+async def test_async_step_reauth_confirm_invalid_auth(hass: HomeAssistant) -> None:
+    """Test reauth confirm step with invalid authentication."""
+    flow = CompitConfigFlow()
```

Comment:
Given how often you mock this, it might be nice to have a separate fixture for this.